app/src/lib/release-notes.ts

```diff
@@ -0,0 +1,105 @@
+const ordinal: (i: number) => string = require('ordinal')
+
+import {
+  ReleaseMetadata,
+  ReleaseNote,
+  ReleaseSummary,
+} from '../models/release-notes'
+
+const itemEntryRe = /^\[(new|fixed|improved|removed|added|pretext)\]\s(.*)/i
+
+const monthNames = [
+  'January',
+  'February',
+  'March',
+  'April',
+  'May',
+  'June',
+  'July',
+  'August',
+  'September',
+  'October',
+  'November',
+  'December',
+]
+
+function formatDate(date: Date) {
+  const day = date.getDate()
+  const dayWithSuffix = ordinal(day)
+  const monthIndex = date.getMonth()
+  const year = date.getFullYear()
+
+  return `${monthNames[monthIndex]} ${dayWithSuffix} ${year}`
+}
+
+export function parseReleaseEntries(
+  notes: ReadonlyArray<string>
+): ReadonlyArray<ReleaseNote> {
+  const entries = new Array<ReleaseNote>()
+
+  for (const note of notes) {
+    const text = note.trim()
+    const match = itemEntryRe.exec(text)
+    if (match === null) {
+      continue
+    }
+    const kind = match[1].toLowerCase()
+    const message = match[2]
+
+    switch (kind) {
+      case 'new':
+      case 'fixed':
+      case 'improved':
+      case 'removed':
+      case 'added':
+      case 'pretext':
+        entries.push({ kind, message })
+    }
```

Comment:
should we log an error if we get an unexpected entry type?

Reply:
I've just added `log.debug` in f16be005c to surface these. We should be verifying these match our expected format earlier (like a CI step).